soc/arm/nxp_s32/s32k/Kconfig.defconfig.series

```diff
@@ -0,0 +1,27 @@
+# S32 K M7 core series
+
+# Copyright 2023 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+if SOC_SERIES_S32K3_M7
+
+config SOC_SERIES
+	default "s32k"
+
+config SYS_CLOCK_HW_CYCLES_PER_SEC
+	default 2000000
+
+config NUM_IRQS
+	# must be >= the highest interrupt number used
+	default 239
+
+if !XIP
```

Comment:
Why is this section setting the flash size to 0 when `CONFIG_XIP=n` required? I see it is also present for the s32ze enablement, but I am not clear why it is needed.

Reply:
These options don't have default values when `!XIP` (https://github.com/zephyrproject-rtos/zephyr/blob/main/arch/Kconfig#L216-L231) and a value is required in many places of the architecture code like Cortex-M/R linker files, MPU code, etc. So a default must be set either at board or SoC level. We chose do it at SoC level so it works for other K3 boards when introduced, like done for instance in https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/arm/mimx8mm_evk/Kconfig.defconfig#L11